docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -826,6 +826,20 @@ const classes = makeStyles(theme => ({
   +<Slider components={{ Thumb: CustomThumb }} />
   ```
 
+- The `component` prop is removed. You should prefer using the `componentsProps`' `root` object.
+
+  ```diff
+  -<Slider component="div" />
+  +<Slider componentsProps={{ root: { as: "div" } }} />
+  ```
```

Comment:
What's the motivation for this breaking change? I was assuming that the `component` prop had the same purpose as the `className` prop, even though there are `classes` and `components`, for simplicity when the objective is to have components as close as possible to the DOM nodes. 

Does the alternative API work well with TypeScript? e.g. adding new props?

Reply:
If you mean that the component prop should do this transformation:
`<Slider component="div" />` =>`<Slider componentsProps={{ root: { as: "div" } }} />`, then we can have it. The problem with my implementation was that it was replacing the whole styled component, which is wrong and not compatible with the previous implementation. If this transformation is ok in your opinion I will add it and keep the `component` prop as just a shorthand for the `componentProps.root.as` prop

Reply:
Reverted the behavior to be as before, I guess it was just a wrong implementation when I converted the component to v5